src/io_threads.c

```diff
@@ -218,16 +219,12 @@ static void *IOThreadMain(void *myid) {
     snprintf(thdname, sizeof(thdname), "io_thd_%ld", id);
     valkey_set_thread_title(thdname);
     serverSetCpuAffinity(server.server_cpulist);
-    makeThreadKillable();
     initSharedQueryBuf();
-    pthread_cleanup_push(freeSharedQueryBuf, NULL);
 
     thread_id = (int)id;
     size_t jobs_to_process = 0;
     IOJobQueue *jq = &io_jobs[id];
-    while (1) {
-        /* Cancellation point so that pthread_cancel() from main thread is honored. */
-        pthread_testcancel();
+    while (io_thread_running[id]) {
```

Comment:
Do we care about the memory fence?

```suggestion
    while (atomic_load_explicit(&io_thread_running[id], memory_order_relaxed)) {
```

Reply:
We don't strictly need to based on the usage, we only transition once from true to false, so it doesn't really matter if it takes longer to recognize the load.